bindgen/TypeTranslator.cpp

```diff
@@ -68,111 +57,143 @@ TypeTranslator::TranslateFunctionPointer(const clang::QualType &qtpe,
     }
 }
 
-std::string TypeTranslator::TranslatePointer(const clang::QualType &pte,
-                                             const std::string *avoid) {
+Type *TypeTranslator::TranslatePointer(const clang::QualType &pte,
+                                       const std::string *avoid) {
 
     if (pte->isBuiltinType()) {
         const clang::BuiltinType *as = pte->getAs<clang::BuiltinType>();
 
         // Take care of void*
         if (as->getKind() == clang::BuiltinType::Void) {
-            return "native.Ptr[Byte]";
+            return new PointerType(new PrimitiveType("Byte"));
         }
 
         // Take care of char*
         if (as->getKind() == clang::BuiltinType::Char_S ||
             as->getKind() == clang::BuiltinType::SChar) {
-            return "native.CString";
+            return new PrimitiveType("native.CString");
         }
     }
 
-    return std::string("native.Ptr[") + Translate(pte, avoid) +
-           std::string("]");
+    return new PointerType(translate(pte, avoid));
 }
 
-std::string
-TypeTranslator::TranslateStructOrUnion(const clang::QualType &qtpe) {
+Type *TypeTranslator::translateStruct(const clang::QualType &qtpe) {
     if (qtpe->hasUnnamedOrLocalType()) {
         // TODO: Verify that the local part is not a problem
         uint64_t size = ctx->getTypeSize(qtpe);
-        return "native.CArray[Byte, " + uint64ToScalaNat(size) + "]";
+        return new ArrayType(new PrimitiveType("Byte"), size);
```

Comment:
Yes, I think so

Reply:
I found the case where it is used:
```c
struct myStruct {
    struct {
        int a;
    } innerStruct;
};
```
Currently this inner struct translates into array of bytes, but I suppose it will be better if we create a `type` for it. I'll create an issue for it

Reply:
👍